05libraries/sec02LibraryBasics.md

```diff
@@ -0,0 +1,113 @@
+---
+title: Library Basics
+---
+
+## Library Basics
+
+### Reviewing the build process
+
+When building an application there are three important steps:
+
+- **preprocessing**: follow the directives (lines started by `#`) on the files to combine the units into what's passed to the compiler;
+- **compilation**: translates the program into machine language code - object files;  and
+- **linking**: merges the various object files and links all the referred libraries as needed to create the executable.
+
+Though normally these steps are invoked by a single command, you can run them one at a time.
+
+- preprocessor:
+
+  ```bash
+  g++ -E -o <output> <input.cpp>
+  ```
+- compilation:
+
+  ```bash
+  g++ -c -o <output.o> <input.cpp>
+  ```
+- linking:
+
+  ```bash
+  g++ -o <executable> <output.o>
+  ```
```

Comment:
We haven't shown anyone `g++` yet, although some of them do know it. But this probably needs a bit of introduction otherwise people might be confused.